src/dasi/core/FileReadHandle.h

```diff
@@ -0,0 +1,61 @@
+
+#pragma once
+
+#include "dasi/api/Key.h"
+#include "dasi/api/ObjectLocation.h"
+#include "dasi/api/ReadHandle.h"
+
+#include "dasi/core/HandleFactory.h"
+
+#include <filesystem>
+#include <fstream>
+
+
+namespace dasi::core {
+
+//----------------------------------------------------------------------------------------------------------------------
+
+class FileReadHandle : public api::ReadHandle {
+
+public: // methods
+
+    FileReadHandle(const std::filesystem::path& path, long long offset = 0);
+    ~FileReadHandle();
+
+    size_t read(void* buf, size_t len) override;
+
+    void open() override;
+    void close() override;
+
+private: // methods
+
+    void print(std::ostream& s) const override;
+
+private: // members
+
+    std::filesystem::path path_;
+    long long offset_;
+    std::ifstream stream_;
```

Comment:
Do we want to do this with all of the buffering, etc, machinery of std::istream?
I would be somewhat tempted to use POSIX apis here.

Reply:
I kept away from them following an eraly discussion where we mentioned that we wanted to use the STL features as much as possible. Happy to switch to POSIX instead.